src/coreComponents/physicsSolvers/fluidFlow/DissipationCompositionalMultiphaseFVMKernels.hpp

```diff
@@ -0,0 +1,395 @@
+    m_omegaDBC( omega ),
+    m_miscibleDBC( miscible )
+  {
+    /// Step 1. Calculate the continuation parameter based on the current Newton iteration
+    m_kappaDBC = 1.0;   // default value
+
+    if( continuation )   // if continuation is enabled
+    {
+      if( curNewton >= 5 )
+      {
+        m_kappaDBC = kappamin;
+      }
+      else
+      {
+        for( int mp = 0; mp < curNewton; mp++ )
+        {
+          m_kappaDBC *= contMultiplier;
+        }
+        m_kappaDBC = std::max( m_kappaDBC, kappamin );
+      }
+    }
+  }
+
+  /**
+   * @brief Compute the local flux contributions to the residual and Jacobian, including dissipation
+   * @param[in] iconn the connection index
+   * @param[inout] stack the stack variables
+   */
+  GEOS_HOST_DEVICE
+  void computeFlux( localIndex const iconn,
+                    typename Base::StackVariables & stack ) const
+  {
+    // ***********************************************
+    // First, we call the base computeFlux to compute:
+    //  1) compFlux and its derivatives,
+    //
+    // We use the lambda below (called **inside** the phase loop of the base computeFlux) to compute dissipation terms
+    Base::computeFlux( iconn, stack, [&] ( integer const ip,
+                                           localIndex const (&k)[2],
+                                           localIndex const (&seri)[2],
+                                           localIndex const (&sesri)[2],
+                                           localIndex const (&sei)[2],
+                                           localIndex const connectionIndex,
+                                           localIndex const k_up,
+                                           localIndex const er_up,
+                                           localIndex const esr_up,
+                                           localIndex const ei_up,
+                                           real64 const & potGrad,
+                                           real64 const & phaseFlux,
+                                           real64 const (&dPhaseFlux_dP)[2],
+                                           real64 const (&dPhaseFlux_dC)[2][numComp] )
+    {
+      GEOS_UNUSED_VAR( k_up, potGrad, phaseFlux, dPhaseFlux_dP, dPhaseFlux_dC, er_up, esr_up, ei_up );
+
+      /// Storing dissipation flux and its derivatives locally
+      real64 dissFlux[numComp]{};
+      real64 dDissFlux_dP[numFluxSupportPoints][numComp]{};
+      real64 dDissFlux_dC[numFluxSupportPoints][numComp][numComp]{};
+      real64 fluxPointCoef[numFluxSupportPoints] = {1.0, -1.0}; // for gradients
+
+      real64 viscosityMult[3] = {1.0, 1.0, 1.0}; // for viscosity
+
+      /// Step 2. Collect all contributions
+      real64 poreVolume_n = 0.0; // Pore volume contribution
+      real64 trans = 0.0; // Transmissibility contribution
+      for( integer ke = 0; ke < numFluxSupportPoints; ++ke )
+      {
+        trans = std::max( stack.transmissibility[connectionIndex][ke], trans );
```

Comment:
This I believe is supposed to bring you sign for your fluxes with pair of positive of negative values depending on norm convention. See. here

https://github.com/GEOS-DEV/GEOS/blob/4d2011e8f50680bb5a7dbeff29f58d6bf8e67239/src/coreComponents/physicsSolvers/fluidFlow/IsothermalCompositionalMultiphaseFVMKernelUtilities.hpp#L151

Reply:
> This I believe is supposed to bring you sign for your fluxes with pair of positive of negative values depending on norm convention. See. here
> 
> https://github.com/GEOS-DEV/GEOS/blob/4d2011e8f50680bb5a7dbeff29f58d6bf8e67239/src/coreComponents/physicsSolvers/fluidFlow/IsothermalCompositionalMultiphaseFVMKernelUtilities.hpp#L151

fixed.